### PR TITLE
chore: update javascript package dependencies

### PR DESCRIPTION
**Pull request by iisakkirotko**
First commit is just bumping nodejs 14 -> 20 to check if build gets the same error as my local machine :)

### CODE REVIEW

**Comment by iisakkirotko** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/widgetti/solara/484?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #479** <a href="https://app.graphite.dev/github/pr/widgetti/solara/479?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #481** <a href="https://app.graphite.dev/github/pr/widgetti/solara/481?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #480** <a href="https://app.graphite.dev/github/pr/widgetti/solara/480?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #483** <a href="https://app.graphite.dev/github/pr/widgetti/solara/483?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #484** <a href="https://app.graphite.dev/github/pr/widgetti/solara/484?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/widgetti/solara/484?utm_source=stack-comment).

File: packages/solara-vuetify-app/package.json [hunk 1]
```diff
@@ -19,14 +19,14 @@
     "material-design-icons-iconfont": "^5.0.1",
     "typeface-roboto": "0.0.54",
     "vue": "~2.6.14",
-    "vuetify": "~2.2.26"
+    "vuetify": "~2.2.34"
```
Comment (maartenbreddels) [inline packages/solara-vuetify-app/package.json:22]:
```suggestion
    "vuetify": "~2.2.26"
```
We need to keep this in sync with https://github.com/widgetti/ipyvuetify/blob/778deaa9c289f9466f421b11f2042b73b251ffc3/js/package.json#L59

File: packages/solara-widget-manager/package.json [hunk 2]
```diff
@@ -6,34 +6,34 @@
...
+    "@lumino/signaling": "^1.11.1",
+    "@lumino/virtualdom": "^1.14.3",
+    "@lumino/widgets": "^1.37.2",
+    "katex": "^0.16.9"
   },
   "devDependencies": {
     "npm-run-all": "^4.1.5",
-    "p-limit": "^2.2.2",
-    "typescript": "~4.1.3"
+    "p-limit": "^5.0.0",
```
Comment (maartenbreddels) [inline packages/solara-widget-manager/package.json:35]:
```suggestion
```

We are not using this anymore

File: packages/solara-widget-manager8/package.json [hunk 3]
```diff
@@ -6,32 +6,32 @@
...
+    "@lumino/virtualdom": "^1.14.3",
+    "@lumino/widgets": "^1.37.2",
+    "katex": "^0.16.9",
     "patch-package": "^8.0.0"
   },
   "devDependencies": {
     "npm-run-all": "^4.1.5",
-    "p-limit": "^2.2.2",
-    "typescript": "~4.1.3"
+    "p-limit": "^5.0.0",
```
Comment (maartenbreddels) [inline packages/solara-widget-manager8/package.json:33]:
```suggestion
```
same

**Review by maartenbreddels — DISMISSED** [review]
just a few changes (i don't think you can accept them, you need to run npm install to also update the lockfile)

File: packages/solara-vuetify-app/package.json (hunk 1 above)
Comment (iisakkirotko) [reply packages/solara-vuetify-app/package.json:22]: We had a discussion about this with @mariobuikhuizen back in the day, and if I remember correctly, thought that there isn't much harm that would come from bumping it up to another compatible patch version. Though maybe it is clearer to keep the versions matching, also as a reminder to bump up the version for ipyvuetify.

**Comment by iisakkirotko** [comment]
> There is still an vuetify version bump, you can remove that and update the lockfile please?

Apparently there is, yeah... The new version matches https://github.com/widgetti/ipyvuetify/blob/7729f015ad384ab1549b25dc21175d3b7f246452/js/package.json#L64, though. Should I change it to what it was instead of that?

**Comment by maartenbreddels** [comment]
That's the vue3/vuetify3 branch: https://github.com/widgetti/ipyvuetify/blob/vuetify3/js/package.json#L64

for vuetify 2 we use mario's branch

**Comment by iisakkirotko** [comment]
@maartenbreddels So then all Vuetify versions match, correct? (I also did run `npm install`, but this produced no change in `package-lock.json`)

Also, if we want to keep our Vuetify versions in sync with ipyvuetify, perhaps we should hard fix them in both, instead of using the compatible with `~` operator? You can see [here](https://github.com/widgetti/solara/blob/b51bcfc5bcd54dda6567b0a9c34fd1bffc831d8f/packages/solara-vuetify-app/package-lock.json#L1852-L1859) that currently npm picks the newest compatible version (i.e. 2.2.34) anyway.

**Pull request merged**